Library/Formula/lua.rb

```diff
@@ -58,6 +58,9 @@ def pour_bottle?
   def install
     ENV.universal_binary if build.universal?
 
+    # Subtitute formula prefix in `src/Makefile` for install name (dylib ID).
+    inreplace "src/Makefile", "@LUA_PREFIX@", prefix
```

Comment:
Almost the same reasoning like above. All other formulae also produce dylibs with a dylib ID that points into `prefix`. The rewrite to `opt_prefix` happens later in `Keg#fix_install_names` (the `opt`-based dylib ID is constructed [here](https://github.com/Homebrew/homebrew/blob/6e5d803878b2ea35b2243ce4fa1192e7d3b5e8a8/Library/Homebrew/keg_relocate.rb#L151-L157)).